src/Aspire.Dashboard/Telemetry/DashboardTelemetrySender.cs

```diff
@@ -0,0 +1,160 @@
+                                }
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            _logger.LogWarning(ex, "Failed to send telemetry request {Name}.", context.Name);
+                        }
+                    }
+                }
+            }
+            finally
+            {
+                _logger.LogInformation("Ending sender loop.");
+            }
+        });
+    }
+
+    private object GetResponseProperty(OperationContextProperty propertyId)
+    {
+        return propertyId.GetValue();
+    }
+
+    public async Task<bool> TryStartTelemetrySessionAsync()
+    {
+        Debug.Assert(_isEnabled is null, "Telemetry session has already been started.");
+
+        _isEnabled = await TryStartTelemetrySessionCoreAsync().ConfigureAwait(false);
+        if (_isEnabled.Value)
+        {
+            StartSendLoop();
+        }
+
+        return _isEnabled.Value;
+    }
+
+    internal bool CreateHttpClient()
+    {
+        if (DebugSessionHelpers.HasDebugSession(_options.Value.DebugSession, out var certificate, out var debugSessionUri, out var token))
+        {
+            if (_options.Value.DebugSession.TelemetryOptOut is not true)
+            {
+                Client = DebugSessionHelpers.CreateHttpClient(debugSessionUri, token, certificate, CreateHandler);
+                return true;
+            }
+        }
+
+        _logger.LogInformation("Telemetry is not configured.");
+        return false;
+    }
+
+    private async Task<bool> TryStartTelemetrySessionCoreAsync()
+    {
+        if (!CreateHttpClient())
+        {
+            return false;
+        }
+
+        Debug.Assert(Client is not null);
+
+        try
+        {
+            var response = await Client.GetAsync(TelemetryEndpoints.TelemetryEnabled).ConfigureAwait(false);
+            var isTelemetryEnabled = response.IsSuccessStatusCode && await response.Content.ReadFromJsonAsync<TelemetryEnabledResponse>().ConfigureAwait(false) is { IsEnabled: true };
+
+            if (!isTelemetryEnabled)
+            {
+                return false;
+            }
+
+            // start the actual telemetry session
+            var telemetryStartedStatusCode = (await Client.PostAsync(TelemetryEndpoints.TelemetryStart, content: null).ConfigureAwait(false)).StatusCode;
+            return telemetryStartedStatusCode is HttpStatusCode.OK;
+        }
+        catch (Exception ex)
+        {
+            // If the request fails, we've been given an invalid server address and should assume telemetry is unsupported.
+            _logger.LogWarning(ex, "Failed to request whether telemetry is supported.");
+            return false;
+        }
+    }
+
+    public void MakeRequest(OperationContext context, Func<HttpClient, Func<OperationContextProperty, object>, Task> requestFunc)
+    {
+        _channel.Writer.TryWrite((context, requestFunc));
```

Comment:
I think a check here is needed for whether the sender is enabled or not. We don't want someone to mistakenly queue requests here and fill up the channel when not enabled.

Probably should throw if not enabled or not initialized